### PR TITLE
[15.0][IMP] helpdesk_mgmt: Add empty option to Team and Category fields to portal ticket form

### PR DESCRIPTION
**Pull request by victoralmau**
Changes done:

- [x] Add empty option to Team and Category fields to portal ticket form.
- [x] Add settings to define required fields (Category and Team)

**Before**
antes

**After**
despues

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT45297

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This should go along with a configuration to force required or not those data.

**Comment by victoralmau** [comment]
Added changes to set fields as required or not, do you think this is the right approach?

File: helpdesk_mgmt/controllers/main.py [hunk 1]
```diff
@@ -47,13 +47,18 @@ def create_new_ticket(self, **kw):
         )
         email = http.request.env.user.email
         name = http.request.env.user.name
+        company = request.env.company
+        team_id_required_field = "helpdesk_mgmt_portal_team_id_required"
+        category_id_required_field = "helpdesk_mgmt_portal_category_id_required"
```
Comment (chienandalu) [inline helpdesk_mgmt/controllers/main.py]: This implementation is a little bit weird, why not getting the value directly in the dict below?

File: helpdesk_mgmt/controllers/main.py [hunk 2]
```diff
@@ -47,13 +47,20 @@ def create_new_ticket(self, **kw):
...
                 "categories": categories,
                 "teams": self._get_teams(),
                 "email": email,
                 "name": name,
+                "ticket_team_id_required": company[
+                    "helpdesk_mgmt_portal_team_id_required"
+                ],
+                "ticket_category_id_required": company[
+                    "helpdesk_mgmt_portal_category_id_required"
+                ],
```
Comment (chienandalu) [inline helpdesk_mgmt/controllers/main.py]: Why not simply:
```suggestion
                "ticket_team_id_required": (
                    company.helpdesk_mgmt_portal_team_id_required
                ),
                "ticket_category_id_required": (
                    company.helpdesk_mgmt_portal_category_id_required
                ),
```

**Review by chienandalu — APPROVED** [review]
:smile: :+1:

**Comment by victoralmau** [comment]
Ping @pedrobaeza

**Review by pedrobaeza — COMMENTED** [review]
The new required should be True by default, as it's the most similar to the previous one (where you should always select one).

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**